guides/creating-a-custom-stepper-using-the-cdk-stepper.md

```diff
@@ -0,0 +1,187 @@
+# Creating a custom stepper using the CDK stepper
+
+Using the [CDK stepper](https://material.angular.io/cdk/stepper/overview) it is possible to build a custom stepper which you can completely style yourself without any specific Material Design styling.
+
+In this guide, we'll learn how we can build our own custom stepper using the CDK stepper. Here is what we'll build by the end of this guide:
+
+<!-- example(cdk-custom-stepper-without-form) -->
+
+## Add CdkStepperModule to your project 
+
+After adding the Angular CDK to your Angular project, the next step is to add the `CdkStepperModule` to your Angular module:
+
+```typescript
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { CdkStepperModule } from '@angular/cdk/stepper'; // this is the relevant import
+
+import { AppComponent } from './app.component';
+
+@NgModule({
+  imports: [BrowserModule, CdkStepperModule], // add the module to your imports
+  declarations: [AppComponent],
+  bootstrap: [AppComponent],
+})
+export class AppModule {}
+```
+
+## Create our custom stepper component
+
+Now we are ready to create our custom stepper component. Therefore, we need to create a new Angular component which extends `CdkStepper`:
+
+__custom-stepper.component.ts__
+```typescript
+import { Directionality } from '@angular/cdk/bidi';
+import { ChangeDetectorRef, Component, Input, QueryList } from '@angular/core';
+import { CdkStep, CdkStepper } from '@angular/cdk/stepper';
+
+@Component({
+  selector: 'app-custom-stepper',
+  templateUrl: './custom-stepper.component.html',
+  styleUrls: ['./custom-stepper.component.css'],
+  providers: [{ provide: CdkStepper, useExisting: CustomStepperComponent }],
```

Comment:
It would be good to add a comment here like
```ts
// This custom stepper provides itself as CdkStepper so that it can be recognized
// by other components.
```

Reply:
Done